src/sql/sql-comparison/sql-comparison.ts

```diff
@@ -276,36 +271,22 @@ export class SqlComparison extends SqlExpression {
   }
 
   public negate(): this {
-    let { op, negated, decorator } = this;
-
-    switch (op) {
-      case '=':
-      case '<>':
-      case '<':
-      case '>':
-      case '<=':
-      case '>=':
-        op = ANTI_OP[op]!;
-        if (decorator) {
-          decorator = decorator === 'ALL' ? 'ANY' : 'ALL';
-        }
-        break;
-
-      default:
-        negated = !negated;
-        break;
-    }
+    const { op, decorator } = this;
 
     const value = this.valueOf();
-    value.op = op;
-    value.negated = negated;
-    if (!value.negated) {
-      value.keywords = this.getKeywordsWithout('not');
+    value.op = ANTI_OP[op];
+    value.keywords = this.getKeywordsWithout('op');
+    if (decorator) {
+      value.decorator = decorator === 'ALL' ? 'ANY' : 'ALL';
     }
-    value.decorator = decorator;
+
     return SqlBase.fromValue(value);
   }
 
+  public hasNot(): boolean {
+    return this.op.includes('NOT');
```

Comment:
I guess we don't have any operators called `NOTHING` or `SNOT` so this is probably fine...

Reply:
I guess if we ever add them then we would change that line to be `/\bNOT\b/.test(this.op)`. I think it is fine for now